luda-adventure/src/app/game/types/game_object/types/positioner/predict_collision.rs

```diff
@@ -121,22 +121,43 @@ fn predict_intersection_on_one_axis(
     target_less_side: Tile,
     target_greater_side: Tile,
 ) -> IntersectionPrediction {
+    enum MovementDirection {
+        Positive,
+        Zero,
+        Negative,
+    }
+    let movement_direction = match (speed * 1.0.ms())
+        .partial_cmp(&0.0_f32.tile())
+        .expect("Invalid speed")
+    {
+        std::cmp::Ordering::Less => MovementDirection::Negative,
+        std::cmp::Ordering::Equal => MovementDirection::Zero,
+        std::cmp::Ordering::Greater => MovementDirection::Positive,
+    };
```

Comment:
I hope `Per` implement [is_sign_negative](https://doc.rust-lang.org/std/primitive.f64.html#method.is_sign_negative) for this, because it looks not cool multiplying 1.0.ms() for calcultation.

Reply:
#379